### PR TITLE
Adds In Social Buttons

### PR DESCRIPTION
**Pull request by veverkap**
This adds social buttons with proper links from the Jekyll data.  This is a pure CSS implementation.

Here is an example of what it looks like:

image

This addresses https://github.com/doomspork/elixir-school/issues/199

### CODE REVIEW

File: _assets/stylesheets/app.scss [hunk 1]
```diff
@@ -1,8 +1,9 @@
 /*
 *= require poole
 *= require syntax
-*= require hyde
```
Comment (veverkap) [inline _assets/stylesheets/app.scss]: Not sure why this changed the line ending.

File: _config.yml [hunk 2]
```diff
@@ -21,6 +21,8 @@ description:
   gr: Μαθήματα προγραμματισμού της γλώσσας Elixir
   it: Lezioni sul linguaggio di programmazione Elixir
 url: https://elixirschool.com
+insecure_url: http://elixirschool.com #Evernote gives a 500 error on the https
```
Comment (veverkap) [inline _config.yml]: There may be an alternative way of doing this, but each time I try to "clip" the https site, it gives an error:

image

File: _assets/stylesheets/app.scss (hunk 1 above)
Comment (doomspork) [inline _assets/stylesheets/app.scss]: No worries, removed extra whitespace. Your editor may automagically do that for you.

**Comment by doomspork** [comment]
@pjhampton / @pragmaticivan / @setzer777 or others have any feedback?  This looks good to me.

Thanks @veverkap :+1:

**Comment by pragmaticivan** [comment]
Nice job @veverkap can you try to simulate with the following css? The current config is really big.

```
*,
*::before,
*::after {
  box-sizing: border-box;
}

// Import icon font, I've used Entypo (http://entypo.com/)
@import url(http://weloveiconfonts.com/api/?family=entypo);

// List of icon unicode symbols from the icon font
// and background colours for the icons
$icon-list: (
    twitter "\F309" #32b9e7,
    facebook "\F30C" #4b70ab,
    googleplus "\F30F" #fa5432,
    pintrest "\F312" #d63533,
    linkedin "\F318" #0087be,
    evernote "\F333" #aaca62,
    email "\2709" #FB7629,
    print "\e716" #167ab6,
);

// Each loop for creating the icon CSS
@each $icon, $unicode, $icon-background in $icon-list {

    .icon--#{$icon} {
        &::before {
            content: $unicode;
        }
    }

    .social__icon--#{$icon} {
      background-color: $icon-background;

        &:hover {
          // Swap black for white to make the icons lighter on hover
          background-color: mix(black, $icon-background, 15%);
        }

    }

}

// Display icons next to each other
.social__item {
    display: inline-block;
    margin-right: 0.4em;
}

// Basic icon style
.icon {
  font-family: 'entypo';
  speak: none;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: 1;

  // Better Font Rendering
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

// Extend icon to all other icons
[class^="icon--"] {
  @extend .icon;
}

// Icon background
.social__icon {
  font-size: 1.2em;
  color: white;
  text-decoration: none;
  // Remove border radius to make icons square
  border-radius:100%;
  width: 1.6em;
  height: 1.6em;
  text-align: center;

  // Vertical Centering
  display: flex;
  align-items: center;
  justify-content: center;
}

// Extend social__icon to all other icons
[class^="social__icon"] {
  @extend .social__icon;
}

// Add padding to icons container
.social__container {
  padding: 0;
  font-size: 1.2em;
}
```

That removes the padding aligning with the title of the share button and reduce the size of the buttons.

**Comment by pragmaticivan** [comment]
I also think the https problem will be gone if you use the share button with the production website, I tested here locally and for me that was the problem, we don't need this new var. the server just need to recognize the correct referer url.

**Comment by doomspork** [comment]
@veverkap could you rebase?  I've enabled Heroku Review Apps so we can sort out the HTTPs :grinning:

**Comment by pjhampton** [comment]
This is beautiful :+1: great job @veverkap

**Comment by veverkap** [comment]
Rebased

**Comment by veverkap** [comment]
Why is the Close button so close to the Comment button? :)

**Comment by veverkap** [comment]
@pragmaticivan I pulled out the insecure_url

**Comment by pragmaticivan** [comment]
Cool! Nice job @veverkap :beers:

**Comment by doomspork** [comment]
I'm going to look into why Heroku won't deploy this branch to a Review App, it's been wonky.

**Comment by doomspork** [comment]
Going to merge, we'll iron out the bugs in production :+1:

**Comment by pragmaticivan** [comment]
The icon fonts have stopped working in production ;/ haha

**Comment by doomspork** [comment]
@veverkap and I just noticed that @pragmaticivan.

High five for the first PR to fix it? :grinning:

**Comment by pragmaticivan** [comment]
It was github cache, looks like it's working now. 
raw

**Comment by pragmaticivan** [comment]
Never mind, It's not working on safari ;/

**Comment by doomspork** [comment]
@pragmaticivan or @veverkap, any ideas on how to fix this?  I haven't looked into it further.

**Comment by veverkap** [comment]
I looked a little. Best idea would be to self host the fonts.

On Monday, April 11, 2016, Sean Callan <notifications@github.com> wrote:

> @pragmaticivan <https://github.com/pragmaticivan> or @veverkap
> <https://github.com/veverkap>, any ideas on how to fix this? I haven't
> looked into it further.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/doomspork/elixir-school/pull/260#issuecomment-208630968>
>


-- 
Sent from Gmail Mobile

**Comment by doomspork** [comment]
Then that is what we shall do!  Is your master branch good to go @veverkap?

**Comment by veverkap** [comment]
I don't think it is presently. I'll look later and see

On Monday, April 11, 2016, Sean Callan <notifications@github.com> wrote:

> Then that is what we shall do! Is your master branch good to go @veverkap
> <https://github.com/veverkap>?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/doomspork/elixir-school/pull/260#issuecomment-208632544>
>


-- 
Sent from Gmail Mobile

**Comment by doomspork** [comment]
Thanks :+1:

**Pull request merged**